types/aws-lambda/trigger/iot-authorizer.d.ts

```diff
@@ -0,0 +1,48 @@
+import {Handler} from "../handler";
+import {PolicyDocument} from "./api-gateway-authorizer";
+
+export type IoTProtocolType = "http" | "mqtt" | "tls";
+
+export type IoTCustomAuthorizerHandler = Handler<IoTCustomAuthorizerEvent, IoTCustomAuthorizerResult>;
+
+export interface IoTProtocolDataTLS {
+    serverName: string, // The server name indication (SNI) host_name string.
+}
+
+export interface IoTProtocolDataHTTP {
+    headers: Record<string, string>,
+    queryString: string,
+}
+
+export interface IoTProtocolDataMQTT {
+    username?: string,
+    password?: string, // A base64-encoded string.
+    clientId: string // Included in the event only when the device sends the value.
```

Comment:
The comment here seems to imply this should be optional? But from what I've done with mqtt I don't think the client id is optional anyway, so I'm rather confused.